### DISCUSSION TITLE
Is Static export  possible to dynamic pages

### DISCUSSION THREAD

**Discussion by arshad-qadri** [vercel/community#3209] [Help] [answered]
### Problem Description

Hi,
I am writing for static export in nextjs 
I have Dynamic website in nextjs we have more thand 50k plus pages
can we do static export for my app 
Is it preffer able or not ?

**Comment by leerob** [comment, marked as answer]
You'll likely want to explore [Incremental Static Regeneration](https://vercel.com/docs/concepts/incremental-static-regeneration), as builds for a 50,000 page site would likely take a really long time 😄

**Reply by arshad-qadri** [reply]
Thanks for your reply 
Can you plz let me know can we deploy .next build or we need to upload our whole code

And second one When Iam using getSaticPaths with getStaticProps and try to create build using this command "next build" then also it is take very long time

**Reply by amyegan** [reply]
The CLI provides an option to build the project in your local environment and deploy from there. You can see the steps for that at  https://vercel.com/docs/cli/deploying-from-cli#deploying-from-local-build-prebuilt 

However, I also think you should consider Lee's recommendation to try [ISR](https://vercel.com/docs/concepts/incremental-static-regeneration/quickstart#). That could save you a lot of time on builds.

**Reply by arshad-qadri** [reply]
Actually we are using AWS for nextjs hoting
So just Please confirm me can we deploy build without upload code anywhere?

**Reply by leerob** [reply]
If you are not using Vercel, please use the Amplify support forums to get your questions routed correctly, or on the Next.js Discussions, please!

Docs: https://nextjs.org/docs/app/building-your-application/deploying/static-exports
